File: epi_judge_java_solutions/epi/LookAndSay.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class LookAndSay {
  @EpiTest(testDataFile = "look_and_say.tsv")

  public static String lookAndSay(int n) {

    String s = "1";
    for (int i = 1; i < n; ++i) {
      s = nextNumber(s);
    }
    return s;
  }

  private static String nextNumber(String s) {
    StringBuilder result = new StringBuilder();
    for (int i = 0; i < s.length(); ++i) {
      int count = 1;
      while (i + 1 < s.length() && s.charAt(i) == s.charAt(i + 1)) {
        ++i;
        ++count;
      }
      result.append(count).append(s.charAt(i));
    }
    return result.toString();
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "LookAndSay.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi_judge_cpp_solutions/look_and_say.cc
#include <string>

#include "test_framework/generic_test.h"

using std::string;
using std::to_string;

string NextNumber(const string& s);

string LookAndSay(int n) {
  string s = "1";
  for (int i = 1; i < n; ++i) {
    s = NextNumber(s);
  }
  return s;
}

string NextNumber(const string& s) {
  string result;
  for (int i = 0; i < size(s); ++i) {
    int count = 1;
    while (i + 1 < size(s) && s[i] == s[i + 1]) {
      ++i, ++count;
    }
    result += to_string(count) + s[i];
  }
  return result;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"n"};
  return GenericTestMain(args, "look_and_say.cc", "look_and_say.tsv",
                         &LookAndSay, DefaultComparator{}, param_names);
}
